lib/xlibc/include/assert.h

```diff
@@ -0,0 +1,39 @@
+#ifndef _XLIBC_ASSERT_H
+#define _XLIBC_ASSERT_H
+
+#ifndef _xlibc_sym2val
+#define _xlibc_sym2val(x) #x
+#endif
+
+#ifndef _xlibc_sym2str
+#define _xlibc_sym2str(x) _xlibc_sym2val(x)
+#endif
```

Comment:
This macro name is bad in my opinion. Is this a standard name anywhere else?
I feel it implies a lookup from some symbol table. When all it does is "stringify" (for lack of a better term).

Reply:
I've named it similar to the macro from PDCLIB (_PDCLIB_symbol2string) which is where I got the idea of how to convert the expression to a string from.
Would you consider renaming it to `_XLIBC_EXPR2STR` (expression instead of symbol, and uppercase to show it's a macro) acceptable?

Reply:
I think it's better to keep traces to the origin: `_XLIBC_symbol2string` and add a comment that this is actually just stringifying expression, but is named after `_PDCLIB_symbol2string`from PDCLIB.

Alternatively, try to fix this upstream? The underscore suggests this was not meant to be used by outside programs, and the name *is* bad. I personally like your expr2str a lot more - but if it means diverging from upstream it's a bad idea (just confuses people: "why did they change this?!").

Consider this nitpicking. Giving your reasoning, I'm fine with the current solution.